app/src/main/java/pm/gnosis/heimdall/data/db/models/GnosisSafeDb.kt

```diff
@@ -27,6 +26,41 @@ data class GnosisSafeDb(
 fun Safe.toDb() = GnosisSafeDb(address)
 fun GnosisSafeDb.fromDb() = Safe(address)
 
+@Entity(
+    tableName = GnosisSafeOwnerDb.TABLE_NAME,
+    foreignKeys = [
+        ForeignKey(
+            entity = GnosisSafeDb::class,
+            parentColumns = [GnosisSafeDb.COL_ADDRESS],
+            childColumns = [GnosisSafeOwnerDb.COL_SAFE_ADDRESS],
+            onUpdate = ForeignKey.NO_ACTION,
+            onDelete = ForeignKey.CASCADE,
+            deferred = true
+        )
+    ],
+    indices = [
+        Index(value = [GnosisSafeOwnerDb.COL_SAFE_ADDRESS])
+    ]
+)
+data class GnosisSafeOwnerDb(
+    @PrimaryKey
```

Comment:
I would use the address as primary key (or if possible the safe since we normally query by the safe address)